packages/svelte/src/internal/client/dev/equality.js

```diff
@@ -16,7 +23,7 @@ export function init_array_prototype_warnings() {
 	array_prototype.indexOf = function (item, from_index) {
 		const index = indexOf.call(this, item, from_index);
 
-		if (index === -1) {
+		if (index === -1 && is_object(item)) {
```

Comment:
What I don't quite understand is why the proxied version would give a different result - is it not updated yet to the latest value?